renderdoc/driver/gl/gl_debug.cpp

```diff
@@ -379,17 +379,29 @@ void GLReplay::InitDebugData()
 
   DebugData.fixedcolFSProg = CreateShaderProgram(empty, fs);
 
-  GenerateGLSLShader(vs, shaderType, "", GetEmbeddedResource(glsl_mesh_vert), glslBaseVer);
-  GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_mesh_frag), glslBaseVer);
-  GenerateGLSLShader(gs, shaderType, "", GetEmbeddedResource(glsl_mesh_geom), glslBaseVer);
+  if(HasExt[ARB_geometry_shader4])
+  {
+    GenerateGLSLShader(vs, shaderType, "", GetEmbeddedResource(glsl_mesh_vert), glslBaseVer);
+    GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_mesh_frag), glslBaseVer);
+    GenerateGLSLShader(gs, shaderType, "", GetEmbeddedResource(glsl_mesh_geom), glslBaseVer);
+
+    DebugData.meshProg = CreateShaderProgram(vs, fs);
+    DebugData.meshgsProg = CreateShaderProgram(vs, fs, gs);
 
-  DebugData.meshProg = CreateShaderProgram(vs, fs);
-  DebugData.meshgsProg = CreateShaderProgram(vs, fs, gs);
+    GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_trisize_frag), glslBaseVer);
+    GenerateGLSLShader(gs, shaderType, "", GetEmbeddedResource(glsl_trisize_geom), glslBaseVer);
 
-  GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_trisize_frag), glslBaseVer);
-  GenerateGLSLShader(gs, shaderType, "", GetEmbeddedResource(glsl_trisize_geom), glslBaseVer);
+    DebugData.trisizeProg = CreateShaderProgram(vs, fs, gs);
+  }
+  else
+  {
+    GenerateGLSLShader(vs, shaderType, "", GetEmbeddedResource(glsl_mesh_vert), glslBaseVer);
+    GenerateGLSLShader(fs, shaderType, "", GetEmbeddedResource(glsl_mesh_frag), glslBaseVer);
 
-  DebugData.trisizeProg = CreateShaderProgram(vs, fs, gs);
+    DebugData.meshProg = CreateShaderProgram(vs, fs);
+    DebugData.meshgsProg = 0;
+    DebugData.trisizeProg = 0;
```

Comment:
Can you also add a debug message for this, so the user is aware the feature is turned off due to lack of hardware support and not just being broken? Similar to line 364 in this file about quad overdraw being missing.